ui/src/components/CopyButton/index.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { useState } from 'react';
+import * as OutlineIcons from '@heroicons/react/24/outline';
+
+import * as Components from '@/components';
+import * as Stores from '@/stores';
+
+type Props = {
+    id?: string;
+    className?: string;
+    textToCopy: string;
+    title: string;
+};
+
+const CopyButton: React.FC<Props> = (props): React.ReactElement => {
+    const [copied, setCopied] = useState(false);
+    const setToast = Stores.useToastStore((state) => state.setToast);
+
+    const confirmationMessage = 'Copied!';
+
+    const handleCopy = async () => {
```

Comment:
This is an async function, but you're using promise chaining. Better to stick to one than mix them